pkg/job/blueocean.go

```diff
@@ -158,6 +158,33 @@ func (c *BlueOceanClient) getGetNodesAPI(option GetNodesOption) string {
 	return api
 }
 
+// ReplayOption contains some options while replaying a PipelineRun.
+type ReplayOption struct {
+	Folders []string
+	Branch  string
+	RunID   string
```

Comment:
The build ID should be an integer instead of a string.

See `"expectedBuildNumber" : 10,`

More details:
* https://github.com/jenkinsci/blueocean-plugin/tree/master/blueocean-rest#replay-a-pipeline-build
* https://github.com/jenkinsci/jenkins/blob/ffeadea891d228406284e6651b2752ecafae2070/core/src/main/java/hudson/model/Run.java#L180

Reply:
Please see:

https://github.com/jenkinsci/blueocean-plugin/blob/6b27be3724c892427b732f30575fdcc2977cfaef/blueocean-rest/src/main/java/io/jenkins/blueocean/rest/model/BlueRun.java#L62-L66

Reply:
As you can see below, the build-id originally is an integer.

https://github.com/jenkinsci/blueocean-plugin/blob/6b27be3724c892427b732f30575fdcc2977cfaef/blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractRunImpl.java#L84
https://github.com/jenkinsci/jenkins/blob/ffeadea891d228406284e6651b2752ecafae2070/core/src/main/java/hudson/model/Run.java#L1078

https://github.com/jenkinsci/blueocean-plugin/blob/6b27be3724c892427b732f30575fdcc2977cfaef/blueocean-pipeline-api-impl/src/test/java/io/jenkins/blueocean/rest/impl/pipeline/RunImplTest.java#L131

So, the build ID can only be an integer.

Reply:
For now, the build ID is always an integer. But for future, it's uncertain.

Reply:
> For now, the build ID is always an integer. But for future, it's uncertain.

It's impossible to change this.

https://github.com/jenkinsci/jenkins/blob/ffeadea891d228406284e6651b2752ecafae2070/core/src/main/java/hudson/model/Run.java#L180

If you still believe the build id can be non-integer. Please create a discussion on https://community.jenkins.io/

Reply:
Please see:

https://github.com/jenkinsci/jenkins/blob/ffeadea891d228406284e6651b2752ecafae2070/core/src/main/java/hudson/model/Run.java#L1072-L1079

Reply:
yes, please see `Integer.toString(number);`

So, it's an integer.

Reply:
Consider this is not a block, I'm going to merge this PR.